net/bridge.go

```diff
@@ -108,3 +174,304 @@ func FindBridgeIP(bridgeName string, subnet *net.IPNet) (net.IP, error) {
 	// No subnet, or none in the required subnet; just return the first one
 	return netdev.CIDRs[0].IP, nil
 }
+
+type BridgeConfig struct {
+	DockerBridgeName string
+	WeaveBridgeName  string
+	DatapathName     string
+	NoFastdp         bool
+	NoBridgedFastdp  bool
+	IsAWSVPC         bool
+	ExpectNPC        bool
+	MTU              int
+	Mac              string
+	Port             int
+}
+
+func CreateBridge(procPath string, config *BridgeConfig) (BridgeType, error) {
+	bridgeType := DetectBridgeType(config.WeaveBridgeName, config.DatapathName)
+
+	if bridgeType == None {
+		bridgeType = Bridge
+		if !config.NoFastdp {
+			bridgeType = BridgedFastdp
+			if config.NoBridgedFastdp {
+				bridgeType = Fastdp
+				config.DatapathName = config.WeaveBridgeName
+			}
+			odpSupported, err := odp.CreateDatapath(config.DatapathName)
+			if err != nil {
+				return None, errors.Wrapf(err, "creating datapath %q", config.DatapathName)
+			}
+			if !odpSupported {
+				bridgeType = Bridge
+			}
+		}
+
+		var err error
+		switch bridgeType {
+		case Bridge:
+			err = initBridge(config)
+		case Fastdp:
+			err = initFastdp(config)
+		case BridgedFastdp:
+			err = initBridgedFastdp(config)
+		default:
+			err = fmt.Errorf("Cannot initialise bridge type %v", bridgeType)
+		}
+		if err != nil {
+			return None, err
+		}
+
+		if err = configureIPTables(config); err != nil {
+			return bridgeType, errors.Wrap(err, "configuring iptables")
+		}
+	}
+
+	if config.IsAWSVPC {
```

Comment:
Yet more subtleties: AWSVPC is not a distinct type; you can have AWSVPC mode in effect when the underlying bridge is either BridgedFastdp or Bridge.
This capability is not currently useful, but if we did have a hybrid overlay/underlay implementation it would be.